parser.py

```diff
@@ -86,31 +97,57 @@ def __rpow__(self, power):
 
 
 class UnaryOp(ArithmeticOp):
-    def __str__(self):
+    def __init__(self, func, exp):
+        super().__init__(func, exp)
+
+    def __repr__(self):
         return "UnaryOp(func={}, exp={})".format(*self.tuple)
 
 
 class BinaryOp(ArithmeticOp):
-    def __str__(self):
+    def __init__(self, func, exp1, exp2):
+        super().__init__(func, exp1, exp2)
+
+    def __repr__(self):
         return "BinaryOp(func={}, exp1={}, exp2={})".format(*self.tuple)
 
 
 class Op:
-    def __init__(self, name, params, wires, string):
+    def __init__(self, name, params, wires):
         self.name = name
         self.params = params
         self.wires = wires
-        self.string = string
 
-    def __str__(self):
-        return self.string
+    def __repr__(self):
+        if self.params:
+            return f"{self.name}({','.join(str(p) for p in self.params)}) {','.join(str(w) for w in self.wires)};"
+
+        return f"{self.name} {','.join(str(w) for w in self.wires)};"
+
+
+class EqualityCondition:
+    def __init__(self, id_, integer):
+        self.id = id_
+        self.integer = integer
 
     def __repr__(self):
-        return self.__str__()
+        return f"{self.id} == {self.integer}"
 
 
+class ConditionalOp(Op):
```

Comment:
Since the specification always has the if condition inline with the gate application, it makes sense to represent it as a composite object, containing both a condition and an operation.